services/virtual-assistant/src/virtual-assistant/routes.py

```diff
@@ -26,39 +31,40 @@ async def health():
 
 @api_blueprint.route("/talk", methods=["POST"])
 @require_identity_header
-async def talk():
-    data = request.get_json()
+@validate_request(TalkRequest)
+@validate_response(TalkResponse, 200)
+@validate_response(TalkRequestError, 400)
+async def talk(data: TalkRequest):
     identity = request.headers.get("x-rh-identity")
     org_id = get_org_id_from_identity(identity)
 
-    if not data:
-        return jsonify({"error": "No request body"}), 400
-
-    try:
-        validated_data = TalkRequest(**data)
-    except ValidationError as e:
-        return jsonify({"errors": e.errors()}), 400
-
     assistant = get_watson_assistant()
```

Comment:
We won't have watson available everywhere. I was thinking into abstracting this part and use a DI approach to fetch a `ConversationProcessor` (it's hard to name things) that could be watson or a hardcoded one for testing.